pennylane/transforms/qcut.py

```diff
@@ -2725,14 +2878,170 @@ def find_and_place_cuts(
          b: ──RY(0.4)──────╰X──│───╰X────────────RY(0.6)──╰┤ ⟨Y ⊗ Z⟩
          1: ──PrepareNode──────╰C───MeasureNode────────────┤
 
+    Alternatively, if all we want to do is to find the optimal way to fit a circuit onto a smaller
+    device, a :class:`~.CutStrategy` can be used to populate the necessary explorations of cutting
+    parameters. As an extreme example, if the only device at our desposal is a 2-qubit device, a
+    simple cut strategy is to simply specify the the ``max_free_wires`` argument (or equivalently
+    directly passing a :class:`~.Device` to the ``device`` argument):
+
+    >>> cut_strategy = qml.transforms.qcut.CutStrategy(max_free_wires=2)
+    >>> print(cut_strategy.get_cut_kwargs(graph))
+     [{'num_fragments': 2, 'imbalance': 0.5714285714285714},
```

Comment:
> Is it possible to cut the above circuit into 2 fragments with at most 2 qubits?

No, but the problem is that, WE know it's not possible since we peeked inside, but kahypar doesn't know that before it does any partitioning. So it is still worth trying to do it (since there does exist 4-wire circuits that CAN be cut into 2 2-wire fragments). All the "probing" parameters populated are only based on the basic circuit info such as number of wires and gates. In short, We don't know what works until we try.

> And how does the `imbalance` parameter relate?

Kahypar is "lazy" in the sense that it tends to find the "easiest" partitioning for a set of constraint. If I were to only loosely bound the `imbalance`, kahypar would give me trivial solutions (e.g. cutting a 100-gate circuit into 98 + 2 fragments). On the other hand, I also can't make the bound too tight since it'll then not be able to find any partitions.

I'm glad you asked this question, since I spent most of my time for this PR working on the best "heuristics" for this imbalance parameter so that it works for most of the circuits.

Reply:
Thanks!